roles/mailserver/tasks/main.yml

```diff
@@ -146,6 +146,39 @@
   when: www_is_wordpress
 
 - block:
+  - name: Set up opencart directory
+    file:
+      path: /mnt/docker/{{ item }}
+      state: directory
+      recurse: yes
+      owner: 82
+      group: 82
+    with_items:
+    - opencart
+    - php
+  - name: Set up opencart nginx site
+    template:
+      src: files/nginx/www-opencart.conf
+      dest: /mnt/docker/nginx/sites-enabled/opencart.conf
+      owner: 991
+      group: 991
+    notify:
+    - restart nginx
+    - restart opencart
+  - name: See if php custom file exists
+    stat:
+      path: /mnt/docker/php/custom.ini
+    register: php_custom
+  - name: Set up empty custom.ini for php modifications
+    file:
+      path: /mnt/docker/php/custom.ini
+      state: touch
+      owner: 82
+      group: 82
+    when: not php_custom.stat.exists
+  when: www_is_opencart
+
```

Comment:
The block above seems like just a cut/paste of the wordpress stanzas. It also repeats the php `custom.ini` setup. Is the PHP `custom.ini` even needed for `opencart`? It is its own self-contained container.

I could be wrong, but I think the only needed piece here is 

```yaml
- name: Set up opencart directory
   file:
     path: /mnt/docker/opencart
     state: directory
     recurse: yes
    owner: 82
    group: 82
- name: Set up opencart nginx site
   template:
     src: files/nginx/www-opencart.conf
     dest: /mnt/docker/nginx/sites-enabled/opencart.conf
     owner: 991
     group: 991
   notify:
   - restart nginx
   - restart opencart
```